pkg/reconciler/ingress/status.go

```diff
@@ -390,8 +397,8 @@ func (m *StatusProber) getGateway(name string) (*v1alpha3.Gateway, error) {
 }
 
 // listGatewayPodsURLs returns a map where the keys are the Gateway Pod IPs and the values are the corresponding
-// URL templates to be probed.
-func (m *StatusProber) listGatewayURLsPerPods(gateway *v1alpha3.Gateway) (map[string][]string, error) {
+// URL templates and the Gateway Pod Port to be probed.
+func (m *StatusProber) listGatewayTargetsPerPods(gateway *v1alpha3.Gateway) (map[string][]probeTarget, error) {
```

Comment:
Isn't it recommended to use a pointer to avoid copy?

Reply:
I don't think using pointers is due here. You only want to avoid copy for fairly big data. In this case we're talking two small'ish strings. It's likely not doing much of a harm but I'd back out the pointers.

A very short and to the point article on the matter: https://medium.com/@vCabbage/go-are-pointers-a-performance-optimization-a95840d3ef85

Tl;dr: Use values (not pointers) unless you need pointer semantics or you've proven that memory copying is the issue.